engine/dlib/src/dlib/buffer.cpp

```diff
@@ -586,4 +588,24 @@ namespace dmBuffer
         *type_count = stream->m_ValueCount;
         return RESULT_OK;
     }
+
+    Result GetContentVersion(HBuffer hbuffer, uint32_t* version)
+    {
+        Buffer* buffer = GetBuffer(g_BufferContext, hbuffer);
+        if (!buffer) {
+            return RESULT_BUFFER_INVALID;
+        }
+        *version = (uint32_t)buffer->m_ContentVersion;
+        return RESULT_OK;
+    }
+
+    Result UpdateContentVersion(HBuffer hbuffer)
+    {
+        Buffer* buffer = GetBuffer(g_BufferContext, hbuffer);
+        if (!buffer) {
+            return RESULT_BUFFER_INVALID;
+        }
+        buffer->m_ContentVersion++;
+        return RESULT_OK;
+    }
```

Comment:
Since there is no buffer lock()/unlock() mechanism, we have to rely on the user notifying the buffer when it needs to update.